qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -57,3 +62,71 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown sample template action: %s' % action)
         self.write(result)
+
+
+def _build_sample_summary(study_id, user_id):
```

Comment:
After some testing, this is the fastest way to build the table. It still takes ~1.5 seconds to build and render on my computer, but it beats the minutes the current one takes (`df.to_html` is really slow) and the javascript version, which is fast on the server but takes a long time to render on the page.